### DISCUSSION TITLE
An awkward array looks different from the one of online document

### DISCUSSION THREAD

**Discussion by Star9daisy** [scikit-hep/awkward#2542] [General]
Hi, I'm just get learning how to use awkward. It confuses me that in [How to create arrays of lists](https://awkward-array.org/doc/main/user-guide/how-to-create-lists.html#from-python-lists), doc shows an awkward array as following:
```
[[1, 2, 3],
 [],
 [4, 5],
 [6],
 [7, 8, 9, 10]]
---------------------
type: 5 * var * int64
```
But in my vscode jupyter notebook or in python cli, it looks like this:
```
<Array [[1, 2, 3], [], [...], [6], [7, 8, 9, 10]] type='5 * var * int64'>
```

I install the latest 2.2.4 version via `pip`.

**Comment by agoose77** [comment]
This is a good spot! The reason for this is that we implement a Jupyter display formatter, which provides more detailed information (the type) in addition to the array contents. You can obtain a similar view with [`array.show(type=True)`](https://awkward-array.org/doc/main/reference/generated/ak.Array.html#ak.Array.show)

**Reply by Star9daisy** [reply]
I have tried it. It shows like this:
```
type: 3 * {
    px: float64,
    py: float64,
    pz: float64,
    E: float64,
    ex: float64
}
[{px: 1.2, py: 3.2, pz: 5.4, E: 2.5, ex: 0.78},
 {px: 32.2, py: 64.2, pz: 543, E: 24.1, ex: 0.35},
 {px: 32.5, py: 63.2, pz: 543, E: 24.6, ex: 0}]
```
It shows more info now.  You're realy helpful!

By the way, saying your answer reminds me that I followed the same document several days before, and I got totally the same output as doc. At that time I though it was too much info printed, so I reinstalled version 1.8 to get back the one line display. I could not know what happened these days so it's impossible to reproduce this. But I guess it's alright.

Love awkward! It's an amazing package! Getting more practice in array-oriented programming these days. Thanks!

**Reply by agoose77** [reply]
You will only see the detailed representation in HTML-enabled Jupyter Frontends. You can see the MIME bundle that Awkward generates with
```python
ipython = get_ipython()
ipython.display_formatter.format(array)
```

:)

**Reply by Star9daisy** [reply]
I just tried in jupyter lab. It did show what the doc shows! Thanks a lot!

**Comment by jpivarski** [comment]
I want to keep all of the discussions open. Issues get closed when they're done, but it's valuable to keep discussions around—even if they're resolved—because they're useful to other people with the same questions.
